generator/default.p9k

```diff
@@ -366,15 +367,28 @@ function __p9k_build_segment_cache() {
 
 ###############################################################
 # @description
-#   Refresh a single item in the cache
+#   Refresh a single item/segment in the cache
 ##
 # @args
-#   $1 string The serialized segment data
+#   $1 segment name
+#   $2 boolean true to trigger prompt rendering
 ##
 function __p9k_refresh_cache_item() {
-  local -a segment_meta=("${(@s:·|·:)1}")
+  local segment="$1"
+  for alignment in left right; do
+    for index in $(p9k::find_in_array "${segment}" "${__P9K_DATA[${alignment}_segments]}"); do
```

Comment:
Generalizing the code from `vi_mode` segment is probably a good idea. 👍  Initially I did not do that, because it may be that a segment already knows it joined state. But having such an internal knowledge is pretty unlikely...